### PR TITLE
Apply `infer_kwarg_from_call()` to more checks

### LINKED ISSUE #7761: `get_argument_from_call` doesn't account for **kwargs
### Bug description

`get_argument_from_call` doesn't account for keyword arguments provided by unpacked kwargs. This is causing both false positives and false negatives

False negative
```python
kw={'mode': 'garbage'}
open(fp, **kw)
```

False positive
```python
kw={'maxsplit', 1}
'asdf'.split('s', **kw)[-1]
```

### Configuration

_No response_

### Command used

```shell
pylint c.py
```

### Pylint output

first example: nothing

### Expected behavior

first example: Expected a warning for `bad-open-mode`

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.13.0-dev0
Python 3.11.0 (v3.11.0:deaf509e8f, Oct 24 2022, 14:43:23) [Clang 13.0.0 (clang-1300.0.29.30)]
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
Closes #7761

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Follow-up to #8728, which created `infer_kwarg_from_call()`. I looked for other places we should use to close #7761.

These mostly solve false negatives for various checks, save for one false positive for `use-maxsplit-arg`. Not worth pulling it out to backport.

I did open a separate issue for #8774, though, because it's a crash fix we could backport.

Closes #7761

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
Just a nit, looks great !

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 1]
```diff
@@ -2228,6 +2230,13 @@ def _check_unnecessary_list_index_lookup(
             # is not redundant, hence we should not report an error.
             return
 
+        # Preserve preliminary_confidence if it was INFERENCE
```
[the remark this answers is not in the document]
Comment (jacobtylerwalls) [reply pylint/checkers/refactoring/refactoring_checker.py:2233]: It should just wait for #7121, when we might have a numeric level we can just compare directly.

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 2]
```diff
@@ -2203,12 +2203,14 @@ def _check_unnecessary_list_index_lookup(
...
 
+        preliminary_confidence = HIGH
         try:
             iterable_arg = utils.get_argument_from_call(
                 node.iter, position=0, keyword="iterable"
             )
         except utils.NoSuchArgumentError:
-            return
+            iterable_arg = utils.infer_kwarg_from_call(node.iter, keyword="iterable")
+            preliminary_confidence = INFERENCE
```
[the remark this answers is not in the document]
Comment (jacobtylerwalls) [reply pylint/checkers/refactoring/refactoring_checker.py:2213]: It is preliminary if the second one turns out to be CONTROL_FLOW, was my thought. But then I should change the second condition to != HIGH.

[the remark this answers is not in the document]
Comment (jacobtylerwalls) [reply pylint/checkers/refactoring/refactoring_checker.py:2213]: Well, that's not quite right either. I guess I'd rather keep preliminary for now.

**Comment by jacobtylerwalls** [comment]
Thanks for the review!

**Pull request merged**